Integration test "coverage_basic.idl" fails

Seems to be an endless recursion in the template expansion. Maybe it is more a bug in IDL-Parser rather than in Fast-DDS-Gen.

```console
$ export GRADLE_USER_HOME=/tmp/gradle-user-home-todelete
$ mkdir -p $GRADLE_USER_HOME
$ git clone --recursive https://github.com/eProsima/Fast-DDS-Gen.git  # tag v2.1.2 at time of writing
Cloning into 'Fast-DDS-Gen'...
...
Submodule path 'thirdparty/idl-parser': checked out 'ad28baf822a40f1aa595b613a6866c91c6051bdd'
$ cd Fast-DDS-Gen
$ ./gradlew -info clean build
Downloading https://services.gradle.org/distributions/gradle-6.4-bin.zip
...
Successfully started process 'Gradle Test Executor 1'

FastDDSGenTest > runTests() STANDARD_OUT
...
    >>> COVERAGE_BASIC TEST: PREPARE...
    Done!
        RESULT: OK!


    >>> COVERAGE_BASIC TEST: GENERATE...
    Executing command: java -jar share/fastddsgen/java/fastddsgen.jar -replace -example CMake -test -d build/test/integration/idls/coverage_basic thirdparty/idl-parser/test/idls/coverage_basic.idl
    Loading templates...
    Processing the file thirdparty/idl-parser/test/idls/coverage_basic.idl...
    Exception in thread "main" java.lang.StackOverflowError
    	at org.antlr.stringtemplate.StringTemplate.getAttributeRenderer(StringTemplate.java:1045)
    	at org.antlr.stringtemplate.StringTemplate.getAttributeRenderer(StringTemplate.java:1045)
    	at org.antlr.stringtemplate.StringTemplate.getAttributeRenderer(StringTemplate.java:1045)
    	at org.antlr.stringtemplate.StringTemplate.getAttributeRenderer(StringTemplate.java:1045)
    	at org.antlr.stringtemplate.StringTemplate.getAttributeRenderer(StringTemplate.java:1045)
    	at org.antlr.stringtemplate.StringTemplate.getAttributeRenderer(StringTemplate.java:1045)
...  many more ...

FastDDSGenTest > runTests() SKIPPED

> Task :test FAILED
:test (Thread[Execution worker for ':' Thread 3,5,main]) completed. Took 2 mins 23.622 secs.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> Process 'Gradle Test Executor 1' finished with non-zero exit value 255
  This problem might be caused by incorrect test process configuration.
  Please refer to the test execution section in the User Manual at https://docs.gradle.org/6.4/userguide/java_testing.html#sec:test_execution
...
BUILD FAILED in 3m 10s
9 actionable tasks: 9 executed
```

--- a/build.gradle
+++ b/build.gradle
@@ -172,4 +172,6 @@ task install(type: InstallPath, group: "Install", description: "Install fastddsg
 
 test {
     useJUnitPlatform()
+    testLogging.showStandardStreams = true
+    systemProperties = System.getProperties()
 }

--- a/src/main/java/com/eprosima/fastcdr/idl/templates/FastCdrCommon.stg
+++ b/src/main/java/com/eprosima/fastcdr/idl/templates/FastCdrCommon.stg
@@ -442,8 +442,8 @@ map_max_serialized_size_element(ctx, typecodeMapElement, var, loopvar) ::= <<
     $elseif(typecodeMapElement.isMapType)$
     $var$ += 4$if(ctx.cdr)$ + eprosima::fastcdr::Cdr::alignment($var$, 4)$endif$;
 
-    $map_max_serialized_size_element(ctx=ctx, typecodeMapElement=typecodeMapElement, var=var, loopvar=ctx.nextLoopVarName)$
-    $elseif(typeCodeMapElement.isType_e)$
+    $map_max_serialized_size(ctx=ctx, typecodeMap=typecodeMapElement, var=var, loopvar=ctx.nextLoopVarName)$
+    $elseif(typecodeMapElement.isType_e)$
     $var$ += 4$if(ctx.cdr)$ + eprosima::fastcdr::Cdr::alignment($var$, 4)$endif$;
 
     $sequence_max_serialized_size(ctx=ctx, typecodeSeq=typecodeMapElement, var=var, loopvar=ctx.nextLoopVarName)$
@@ -473,7 +473,7 @@ $elseif(typecodeMapElement.isType_d)$
 $elseif(typecodeMapElement.isMapType)$
     $var$ += 4$if(ctx.cdr)$ + eprosima::fastcdr::Cdr::alignment($var$, 4)$endif$;
 
-    $map_serialized_size_element(ctx=ctx, typecodeMapElement=typecodeMapElement, data=data, var=var, loopvar=ctx.nextLoopVarName)$
+    $map_serialized_size(ctx=ctx, typecodeMap=typecodeMapElement, data=data, var=var, loopvar=ctx.nextLoopVarName)$
 $elseif(typecodeMapElement.isType_e)$
     $var$ += 4$if(ctx.cdr)$ + eprosima::fastcdr::Cdr::alignment($var$, 4)$endif$;
 
@@ -481,7 +481,7 @@ $elseif(typecodeMapElement.isType_e)$
 $elseif(typecodeMapElement.isType_f)$
     $array_serialized_size(ctx=ctx, typecodeArr=typecodeMapElement, data=data, var=var, loopvar=ctx.nextLoopVarName, dimensions=typecodeMapElement.dimensions)$
 $else$
-    $var$ += $typecodeMapElement.scopedname$::getCdrSerializedSize($loopvar$, $var$);
+    $var$ += $typecodeMapElement.scopedname$::getCdrSerializedSize($data$, $var$);
 $endif$
 >>
 

